moshi/src/main/java/com/squareup/moshi/JsonWriter.java

```diff
@@ -386,6 +388,57 @@ public final JsonWriter value(BufferedSource source) throws IOException {
   @CheckReturnValue
   public abstract BufferedSink valueSink() throws IOException;
 
+  /**
+   * Encodes the value which may be a string, number, boolean, null, map, or list.
+   *
+   * @return this writer.
+   * @see JsonReader#readJsonValue()
+   */
+  public final JsonWriter jsonValue(@Nullable Object value) throws IOException {
+    if (value instanceof Map<?, ?>) {
+      beginObject();
+      for (Map.Entry<?, ?> entry : ((Map<?, ?>) value).entrySet()) {
+        Object key = entry.getKey();
+        if (!(key instanceof String)) {
+          throw new IllegalArgumentException(
```

Comment:
what does this do if key is null?

Reply:
NPE creating message. Will fix and test later today. Should also test arrays and maps with null values.